src/rust/engine/fs/src/lib.rs

```diff
@@ -573,9 +616,10 @@ pub trait VFS<E: Send + Sync + 'static>: Clone + Send + Sync + 'static {
   ///
   /// TODO: Should handle symlink loops (which would exhibit as an infinite loop in expand).
   ///
-  fn canonicalize(&self, symbolic_path: PathBuf, link: Link) -> BoxFuture<Option<PathStat>, E> {
+  fn canonicalize(&self, link: Link) -> BoxFuture<Option<PathStat>, E> {
```

Comment:
As far as I can tell `symbolic_path` is always identical to `link.0`

If this isn't correct, please let me know how to populate `symbolic_path` in my new caller :)

Reply:
See the failing unit test in https://travis-ci.org/pantsbuild/pants/jobs/374986336:
```
E   AssertionError: Lists differ: ['a/4.txt.ln'] != [u'a/4.txt.ln', u'd.ln/4.txt.l...
E   
E   Second list contains 1 additional elements.
E   First extra element 1:
E   d.ln/4.txt.ln
E   
E   - ['a/4.txt.ln']
E   + [u'a/4.txt.ln', u'd.ln/4.txt.ln']
_ FSTest.test_walk_recursive_trailing_doublestar _
```